horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -0,0 +1,80 @@
+package io.hyperfoil.tools.horreum.svc;
+
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.junit.jupiter.api.TestInfo;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.hyperfoil.tools.horreum.api.RunService;
+import io.hyperfoil.tools.horreum.entity.json.DataSet;
+import io.hyperfoil.tools.horreum.entity.json.Schema;
+import io.hyperfoil.tools.horreum.entity.json.Test;
+import io.hyperfoil.tools.horreum.server.CloseMe;
+import io.hyperfoil.tools.horreum.server.RoleManager;
+import io.hyperfoil.tools.horreum.test.NoGrafanaProfile;
+import io.hyperfoil.tools.horreum.test.PostgresResource;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.quarkus.test.oidc.server.OidcWiremockTestResource;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+
+@QuarkusTest
+@QuarkusTestResource(PostgresResource.class)
+@QuarkusTestResource(OidcWiremockTestResource.class)
+@TestProfile(NoGrafanaProfile.class)
+public class RunServiceTest extends BaseServiceTest {
+
+   @Inject
+   EntityManager em;
+
+   @Inject
+   RoleManager roleManager;
+
+   @Inject
+   RunService runService;
+
+   @Inject
+   Vertx vertx;
+
+   @org.junit.jupiter.api.Test
+   public void testDataSetCreated(TestInfo info) throws InterruptedException {
+      Test test = createTest(createExampleTest(getTestName(info)));
+      Schema schema = createExampleSchema(info);
+      setTestVariables(test, "Value", "value");
+
+      int runId = uploadRun(runWithValue(schema, 42).toString(), test.name);
+
+      try (CloseMe ignored = roleManager.withRoles(em, Arrays.asList(TESTER_ROLES))) {
+         vertx.executeBlocking(Promise::complete, true).toCompletionStage().toCompletableFuture().get(10, TimeUnit.SECONDS);
```

Comment:
I guess this is supposed to schedule a task in the blocking pool and wait until it's complete, hoping for proper ordering. However the SqlServiceImpl is *not* using ordered execution, and even with that there'd be a race condition.

I believe that a better approach would be publishing an event about DataSet creation on the eventbus (other components will likely wait for that anyway) and waiting for that event here.

Reply:
I'll re-work this as suggested.